data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/IndexParametersConfiguration.java

```diff
@@ -5,30 +5,50 @@
 
 package org.opensearch.dataprepper.plugins.source.opensearch.configuration;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
 import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
 
 public class IndexParametersConfiguration {
 
+    private static final Logger LOG = LoggerFactory.getLogger(IndexParametersConfiguration.class);
+
     @JsonProperty("include")
-    private List<String> include;
+    @Valid
+    private List<IncludedIndex> include = Collections.emptyList();
```

Comment:
@cmanning09 and I had discussed this. The reason we make the include list an object is because there may be specific requirements for each index. Right now, we have a global query, but we may have an ask to have index specific queries, as well as some other things that should be defined at the index level. If we kept with this config you pasted above, we would be making a breaking change. So the options are

1. Keep as is
2. Use list of regex patterns only and very likely introduce a breaking change later
3. Keep it consistent with 
```
include:
          - index_name_regex: "logs[a-z].*"
        exclude:
          - index_name_regex: "bad-.*"
```

Reply:
> If we kept with this config you pasted above, we would be making a breaking change. So the options are

How will it be a breaking change? Sorry, is this source already released?

Reply:
It will be a breaking change if we release this in 2.4

```
  include:
          - "logs[a-z].*"
        exclude:
          - "bad-.*"
```

and then need to migrate to something like this

```
include:
  - index_name_regex: "logs[a-z].*"
     query: "query-for-index"
  - index_name_regex: "logs[a-z].*"
     query: "query-for-second-index"
exclude:
  - "bad-.*"
```

Reply:
Interesting.

Can we rename `index_name_regex` to just `index_name`? OpenSearch does this.

Also, do we want it to be a regex? I think we could follow the pattern that OpenSearch supports which may not be a true regex.

Reply:
We will support regex as is for now and can revisit regular index patterns in the future